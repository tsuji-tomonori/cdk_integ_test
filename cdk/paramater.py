paramater = {
    "lambda": {
        "hello": {
            "env": {
                "LOG_LEVEL": "INFO",
            },
            "memory_size": 128,
        },
    },
}


def build_name(service: str, hostname: str) -> str:
    return f"integ-{service}-{hostname}"
