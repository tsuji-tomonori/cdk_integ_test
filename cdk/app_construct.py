from typing import Any, Self

from constructs import Construct

from cdk.lmd_construct import LambdaConstruct


class AppConstruct(Construct):
    def __init__(
        self: Self,
        scope: Construct,
        construct_id: str,
        **kwargs: Any,  # noqa: ANN401
    ) -> None:
        super().__init__(scope, construct_id, **kwargs)

        self.hello = LambdaConstruct(self, "hello")
