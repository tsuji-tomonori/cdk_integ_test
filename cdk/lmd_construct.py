from pathlib import Path
from typing import Any, Self

import aws_cdk as cdk
from aws_cdk import aws_lambda as lambda_
from aws_cdk import aws_logs as logs
from constructs import Construct

from cdk.paramater import build_name, paramater


class LambdaConstruct(Construct):
    def __init__(
        self: Self,
        scope: Construct,
        construct_id: str,
        **kwargs: Any,  # noqa: ANN401
    ) -> None:
        super().__init__(scope, construct_id, **kwargs)

        self.function = lambda_.Function(
            scope=self,
            id="function",
            code=lambda_.Code.from_asset(
                str(Path.cwd() / "src"),
            ),
            handler="lambda_function.lambda_handler",
            runtime=lambda_.Runtime.PYTHON_3_11,
            environment=paramater["lambda"][construct_id]["env"],  # type: ignore  # noqa: PGH003, E501
            memory_size=paramater["lambda"][construct_id]["memory_size"],  # type: ignore  # noqa: PGH003, E501
            function_name=build_name("lambda", construct_id),
        )

        self.function.add_environment(
            "POWERTOOLS_SERVICE_NAME",
            construct_id,
        )

        self.logs = logs.LogGroup(
            scope=self,
            id="logs",
            log_group_name=f"/aws/lambda/{self.function.function_name}",
            retention=logs.RetentionDays.THIRTEEN_MONTHS,
            removal_policy=cdk.RemovalPolicy.DESTROY,
        )
