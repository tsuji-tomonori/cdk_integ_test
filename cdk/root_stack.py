from typing import Any, Self

from aws_cdk import Stack
from constructs import Construct

from cdk.app_construct import AppConstruct


class RootStack(Stack):
    def __init__(
        self: Self,
        scope: Construct,
        construct_id: str,
        **kwargs: Any,  # noqa: ANN401
    ) -> None:
        super().__init__(scope, construct_id, **kwargs)

        self.app = AppConstruct(self, "app")
