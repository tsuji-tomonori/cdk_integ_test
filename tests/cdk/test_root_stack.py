import json

import aws_cdk as cdk
from aws_cdk import assertions
from aws_cdk import integ_tests_alpha as integ_tests
from syrupy.matchers import path_type

from cdk.root_stack import RootStack


def test_snapshot(snapshot) -> None:
    app = cdk.Stack()
    stack = RootStack(app, "test")
    template = assertions.Template.from_stack(stack)

    matcher = path_type(
        {
            r".*\.S3Key$": (str,),
        },
        regex=True,
    )
    assert template.to_json() == snapshot(matcher=matcher)


def test_integ() -> None:
    app = cdk.Stack()
    stack = RootStack(app, "test")
    integ = integ_tests.IntegTest(
        scope=app,
        id="integ_test",
        test_cases=[stack],
        regions=[stack.region],
    )

    invoke = integ.assertions.invoke_function(
        function_name=stack.app.hello.function.function_name,
    )
    invoke.expect(
        integ_tests.ExpectedResult.object_like(
            expected={
                "statusCode": 200,
                "headers": {
                    "Content-Type": "application/json",
                    "Access-Control-Allow-Origin": "*",
                    "Access-Control-Allow-Methods": "GET, POST, DELETE",
                    "Access-Control-Allow-Credentials": True,
                    "Access-Control-Allow-Headers": "origin, x-requested-with",
                },
                "body": json.dumps(
                    {
                        "message": "hello",
                    },
                ),
                "isBase64Encoded": False,
            },
        ),
    )
