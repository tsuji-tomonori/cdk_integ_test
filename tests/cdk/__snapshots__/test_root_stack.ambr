# serializer version: 1
# name: test_snapshot
  dict({
    'Parameters': dict({
      'BootstrapVersion': dict({
        'Default': '/cdk-bootstrap/hnb659fds/version',
        'Description': 'Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]',
        'Type': 'AWS::SSM::Parameter::Value<String>',
      }),
    }),
    'Resources': dict({
      'apphellofunction1BC4953D': dict({
        'DependsOn': list([
          'apphellofunctionServiceRole2C74464E',
        ]),
        'Properties': dict({
          'Code': dict({
            'S3Bucket': dict({
              'Fn::Sub': 'cdk-hnb659fds-assets-${AWS::AccountId}-${AWS::Region}',
            }),
            'S3Key': str,
          }),
          'Environment': dict({
            'Variables': dict({
              'LOG_LEVEL': 'INFO',
              'POWERTOOLS_SERVICE_NAME': 'hello',
            }),
          }),
          'FunctionName': 'integ-lambda-hello',
          'Handler': 'lambda_function.lambda_handler',
          'MemorySize': 128,
          'Role': dict({
            'Fn::GetAtt': list([
              'apphellofunctionServiceRole2C74464E',
              'Arn',
            ]),
          }),
          'Runtime': 'python3.11',
        }),
        'Type': 'AWS::Lambda::Function',
      }),
      'apphellofunctionServiceRole2C74464E': dict({
        'Properties': dict({
          'AssumeRolePolicyDocument': dict({
            'Statement': list([
              dict({
                'Action': 'sts:AssumeRole',
                'Effect': 'Allow',
                'Principal': dict({
                  'Service': 'lambda.amazonaws.com',
                }),
              }),
            ]),
            'Version': '2012-10-17',
          }),
          'ManagedPolicyArns': list([
            dict({
              'Fn::Join': list([
                '',
                list([
                  'arn:',
                  dict({
                    'Ref': 'AWS::Partition',
                  }),
                  ':iam::aws:policy/service-role/AWSLambdaBasicExecutionRole',
                ]),
              ]),
            }),
          ]),
        }),
        'Type': 'AWS::IAM::Role',
      }),
      'apphellologs3250E3D9': dict({
        'DeletionPolicy': 'Delete',
        'Properties': dict({
          'LogGroupName': dict({
            'Fn::Join': list([
              '',
              list([
                '/aws/lambda/',
                dict({
                  'Ref': 'apphellofunction1BC4953D',
                }),
              ]),
            ]),
          }),
          'RetentionInDays': 400,
        }),
        'Type': 'AWS::Logs::LogGroup',
        'UpdateReplacePolicy': 'Delete',
      }),
    }),
    'Rules': dict({
      'CheckBootstrapVersion': dict({
        'Assertions': list([
          dict({
            'Assert': dict({
              'Fn::Not': list([
                dict({
                  'Fn::Contains': list([
                    list([
                      '1',
                      '2',
                      '3',
                      '4',
                      '5',
                    ]),
                    dict({
                      'Ref': 'BootstrapVersion',
                    }),
                  ]),
                }),
              ]),
            }),
            'AssertDescription': "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
          }),
        ]),
      }),
    }),
  })
# ---
